TritonSE/YSC-Mobile-Application
# Add authentication to initial socket connections

### Administrative Info
[GitHub Issue #10 ]
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

- **Notes**:
  - There are `.env` files for both frontend and backend, will send details in Slack.
  - Console logs are only for testing purposes. Will be removed after reviewers approve PR.

### Changes
What changes did you make?
- Validate token on login/socket connection by making POST request to YSC validate route
  - If invalid token, disconnect socket and redirect user to login screen
- Create `.env` file for backend using dotenv package
  - Created `constants.ts` file to import any env variables

### Testing
How did you confirm your changes worked? 
- Logged in with valid token and confirmed socket connects
- Logged in with invalid token and confirmed socket disconnects and user gets redirected to home screen


### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

| Backend console logs with VALID token |
| --- |
| ![image](https://user-images.githubusercontent.com/64985802/163305679-18445a70-86c5-4c3a-a57e-2707d2e0d1eb.png) |

| Frontend console logs with VALID token |
| --- |
| ![image](https://user-images.githubusercontent.com/64985802/163305805-27b90a8a-ecdc-4757-8727-c5eb8c6b349f.png) |

**Steps for testing invalid token:**

1. Uncomment `invalidToken` in `backend\src\utils\auth.ts` and pass it into authorization header like this ``Authorization: `Bearer ${invalidToken}` ``
2. Log in with test credentials
3. Confirm you get redirected back to login screen (due to invalid token)

| Backend console logs with INVALID token |
| --- |
| ![image](https://user-images.githubusercontent.com/64985802/163306468-234aff37-9a6d-4477-927e-bbeb96ab9f47.png) |

| Frontend console logs with INVALID token |
| --- |
| ![image](https://user-images.githubusercontent.com/64985802/163306533-4b27c0ea-5687-4a19-9466-e2487d7a8002.png) |
